src/Nri/Ui/Carousel/V2.elm

```diff
@@ -0,0 +1,175 @@
+module Nri.Ui.Carousel.V2 exposing
+    ( viewWithCombinedControls, viewWithPreviousAndNextControls
+    , viewWithTabControls, AriaLabel(..)
+    )
+
+{-| Patch changes:
+
+  - added new carousel APIs (with tabbed controls/previous and next controls, and combined controls)
+
+@docs viewWithCombinedControls, viewWithPreviousAndNextControls
+@docs viewWithTabControls, AriaLabel
+
+-}
+
+import Accessibility.Styled.Aria as Aria
+import Css exposing (..)
+import Html.Styled as Html exposing (..)
+import Html.Styled.Attributes as Attrs exposing (css)
+import TabsInternal.V2 as TabsInternal
+
+
+{-| Builds a carousel with previous and next controls
+Returns:
+`controls`: previous and next buttons element
+`slides` the container with the carousel contents
+`containerAttributes` attributes that should be used on the parent div of both the button and slides elements
+-}
+viewWithPreviousAndNextControls :
+    { selected : id
+    , panels :
+        List
+            { id : id
+            , slideHtml : Html msg
+            , ariaLabel : AriaLabel
+            }
+    , viewPreviousButton : Html msg
+    , viewNextButton : Html msg
+    , ariaLabel : AriaLabel
+    , controlListStyles : List Style
+    }
+    ->
+        { controls : Html msg
+        , slides : Html msg
+        , containerAttributes : List (Attribute msg)
+        }
+viewWithPreviousAndNextControls config =
+    { controls =
+        div [ css config.controlListStyles ]
+            [ config.viewPreviousButton, config.viewNextButton ]
+    , slides =
+        List.map
+            (\panel ->
+                Html.div
+                    [ Attrs.attribute "role" "group"
+                    , Aria.roleDescription "slide"
+                    , ariaLabelToAttr panel.ariaLabel
+                    , css
+                        [ if config.selected == panel.id then
+                            Css.display Css.block
+
+                          else
+                            Css.display Css.none
+                        ]
+                    ]
+                    [ panel.slideHtml ]
+            )
+            config.panels
+            |> Html.div [ Attrs.attribute "atomic" "false", Attrs.attribute "live" "polite" ]
+    , containerAttributes =
+        [ Attrs.attribute "role" "region"
```

Comment:
💭 I think we may want to support role=group here, too:

> A carousel container element that encompasses all components of the carousel, including both carousel controls and slides, has either role [region](https://w3c.github.io/aria/#region) or role [group.](https://w3c.github.io/aria/#group) The most appropriate role for the carousel container depends on the information architecture of the page. See the [Landmark Regions Practice](https://www.w3.org/WAI/ARIA/apg/practices/landmark-regions/) to determine whether the carousel warrants being designated as a landmark region.

Worth checking with the bats, but after reading a bit i suspect region makes sense for our use case but may not be the case for Growth.